substrate/frame/delegated-staking/src/lib.rs

```diff
@@ -557,8 +550,6 @@ impl<T: Config> Pallet<T> {
 		let delegator = delegator.get();
 
 		let mut ledger = AgentLedger::<T>::get(&agent).ok_or(Error::<T>::NotAgent)?;
-		// try to hold the funds.
-		T::Currency::hold(&HoldReason::StakingDelegation.into(), &delegator, amount)?;
 
 		let new_delegation_amount =
 			if let Some(existing_delegation) = Delegation::<T>::get(&delegator) {
```

Comment:
I can only increase my delegation amount here it seems. How can I reduce it? by `un_delegating` I suppose?

Reply:
There is a separate `release` function for un_delegating. This code path is only used when creating or adding to existing delegation.